frontend/appflowy_tauri/src/appflowy_app/stores/reducers/document/async_actions/split.ts

```diff
@@ -0,0 +1,54 @@
+import { BlockType, TextDelta } from '@/appflowy_app/interfaces/document';
+import { DocumentController } from '@/appflowy_app/stores/effects/document/document_controller';
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { nanoid } from 'nanoid';
+import { documentActions, DocumentState } from '../slice';
+
+export const splitNodeThunk = createAsyncThunk(
+  'document/splitNode',
+  async (
+    payload: { id: string; retain: TextDelta[]; insert: TextDelta[]; controller: DocumentController },
+    thunkAPI
+  ) => {
+    const { id, controller, retain, insert } = payload;
+    const { dispatch, getState } = thunkAPI;
+    const state = (getState() as { document: DocumentState }).document;
+    const node = state.nodes[id];
+    if (!node.parent) return;
+    const children = state.children[node.children];
+    const prevId = children.length > 0 ? null : node.id;
+    const parent = children.length > 0 ? node : state.nodes[node.parent];
+    const newNode = {
+      id: nanoid(10),
```

Comment:
It would be better if you create a function called gen_node_id, gen_children_node_id. No need  to remember the length of the id